nats/store.go

```diff
@@ -0,0 +1,142 @@
+package nats
+
+import (
+	"errors"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+type BucketManager interface {
+	CreateBucketOrFindExisting(name string) (*bucket, error)
+	BucketExists(name string) (bool, error)
+	DeleteBucket(name string) error
+}
+
+type Bucket interface {
+	GetValue(key string) ([]byte, error)
+	GetKeyCreateDate(key string) (time.Time, error)
+	GetAllTheKeys() ([]string, error)
+	GetHistoryOfTheKey(key string) ([]KeyVal, error)
+	AddPair(key string, val []byte) error
+	Delete(key string) error
+}
+
+type bucketManager struct {
```

Comment:
could we make it exportable?

Reply:
do we need it thought? we have interface BucketManager